File: Api/Models/TblCliente.cs
namespace Api.Models
{
    public class TblCliente
    {
        // Primary Key
        public int Id { get; set; }
        // Properties
        public required string? RazonSocial { get; set; }
        public required int IdTipoCliente { get; set; }
        public required string? RFC { get; set; }
        public DateTime FechaDeCreacion { get; set; }
        // Timestamps
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
        // Foreign Keys
        public CatTipoCliente? TipoCliente { get; set; }
    }

}

File: App/src/app/domain/models/tbl-cliente/tbl-cliente.ts
import { CatTipoCliente } from "../cat-tipo-cliente/cat-tipo-cliente";

export interface TblCliente {
  id: number;
  razonSocial: string;
  idTipoCliente: number;
  rfc: string;
  fechaDeCreacion: Date;
  createdAt: Date;
  updatedAt: Date;
  deletedAt: Date | null;
  tipoCliente: CatTipoCliente;
}
